packages/hasher/src/hashOfFiles.ts

```diff
@@ -13,29 +12,39 @@ import { RepoInfo } from "./repoInfo";
  * Note: We have to force the types because globby types are wrong
  *
  * @param packageRoot The root of the package
- * @param globs Globs inside a package root to consider as part of the hash
- * @param logger An instance of backfill logger
  * @param repoInfo The repoInfo that carries information about repo-wide hashes
  */
 export async function generateHashOfFiles(
   packageRoot: string,
   repoInfo: RepoInfo
 ): Promise<string> {
-  const { repoHashes, root } = repoInfo;
+  const { repoHashes, root, packageHashes } = repoInfo;
 
-  const normalized = path.normalize(packageRoot) + sep;
+  const hashes: string[] = [];
+  const packageRelativeRoot = path.relative(root, packageRoot);
 
-  const files: string[] = Object.keys(repoHashes).filter((f) =>
-    path.join(root, f).includes(normalized)
-  );
+  if (packageHashes[packageRelativeRoot]) {
+    // Fast path: if files are clearly inside a package as per the packageHashes cache
+    for (const hash of packageHashes[packageRelativeRoot]) {
+      hashes.push(hash[0], hash[1]);
+    }
+    return hashStrings(hashes);
+  } else {
+    // Slow old path: if files are not clearly inside a package (mostly the case for malformed monorepos, like tests)
+    const normalized = path.normalize(packageRoot) + sep;
 
-  files.sort((a, b) => a.localeCompare(b));
+    const files: string[] = Object.keys(repoHashes).filter((f) =>
+      path.join(root, f).includes(normalized)
+    );
 
-  const hashes: string[] = [];
+    files.sort((a, b) => a.localeCompare(b));
```

Comment:
Add a comment for the purpose of this sort. I think it's so that your hashes are in a predictable order.